crates/bevy_ecs/src/component.rs

```diff
@@ -283,22 +284,48 @@ impl ComponentInfo {
 /// one `World` to access the metadata of a `Component` in a different `World` is undefined behavior
 /// and must not be attempted.
 #[derive(Debug, Copy, Clone, Hash, Ord, PartialOrd, Eq, PartialEq)]
-pub struct ComponentId(usize);
+pub struct ComponentId {
+    index: u32,
```

Comment:
Doc strings on these fields please.

The intent is important here and these are non-trivial details, even if they're internal.